ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/resources/DistributionPatcher.scala

```diff
@@ -0,0 +1,57 @@
+package ch.epfl.bluebrain.nexus.ship.resources
+
+import cats.effect.IO
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.kernel.Logger
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.FileSelf
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.defaultS3StorageId
+import ch.epfl.bluebrain.nexus.delta.rdf.utils.UriUtils
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, ResourceUris}
+import ch.epfl.bluebrain.nexus.ship.ProjectMapper
+import ch.epfl.bluebrain.nexus.ship.resources.DistributionPatcher._
+import io.circe.Json
+import io.circe.optics.JsonPath.root
+import io.circe.syntax.KeyOps
+
+final class DistributionPatcher(fileSelfParser: FileSelf, projectMapper: ProjectMapper, targetBase: BaseUri) {
+
+  /**
+    * Distribution may be defined as an object or as an array in original payloads
+    */
+  def singleOrArray: Json => IO[Json] = root.distribution.json.modifyA { json =>
+    json.asArray match {
+      case Some(array) => array.traverse(single).map(Json.arr(_: _*))
+      case None        => single(json)
+    }
+  }(_)
+
+  def single: Json => IO[Json] = (json: Json) => fileContentUrl(json).map(toS3Location)
```

Comment:
This method is never used in real code, the tests can easily call the other one

Reply:
It would still be more complicated as an extra level would be required